daemon/src/wallet.rs

```diff
@@ -287,27 +249,33 @@ impl Actor {
         let fee_rate = msg.fee.unwrap_or_else(FeeRate::default_min_relay_fee);
         let address = msg.address;
 
-        let amount = if let Some(amount) = msg.amount {
-            amount
-        } else {
-            self.max_giveable(address.script_pubkey().len(), fee_rate)
-                .context("Unable to drain wallet")?
-        };
+        let mut psbt = {
+            let mut tx_builder = self.wallet.build_tx();
 
-        tracing::info!(%amount, %address, "Amount to be sent to address");
+            tx_builder
+                .fee_rate(fee_rate)
+                // Turn on RBF signaling
+                .enable_rbf();
 
-        let mut tx_builder = self.wallet.build_tx();
+            match msg.amount {
+                Some(amount) => {
+                    tracing::info!(%amount, %address, "Withdrawing from wallet");
 
-        tx_builder
-            .add_recipient(address.script_pubkey(), amount.as_sat())
-            .fee_rate(fee_rate)
-            // Turn on RBF signaling
-            .enable_rbf();
+                    tx_builder.add_recipient(address.script_pubkey(), amount.as_sat());
+                }
+                None => {
+                    tracing::info!(%address, "Draining wallet");
 
-        let (mut psbt, _) = tx_builder.finish()?;
+                    tx_builder.drain_to(address.script_pubkey());
```

Comment:
shouldn't we call `drain_wallet()` before `drain_to()`? (at least that's what the docs say: https://docs.rs/bdk/0.15.0/bdk/wallet/tx_builder/struct.TxBuilder.html#method.drain_to )

Reply:
Good find! Should probably yes. Also, I should probably be testing this :D

Reply:
Tested now and works like a charm! https://mempool.space/testnet/tx/539f544898d2e2d0ae8289dc0677ad0df97eb025a9319a5c8a2c5f46723d52b8